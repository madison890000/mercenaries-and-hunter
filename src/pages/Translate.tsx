import React, { useState } from 'react';
import { Col, Row } from 'antd';
import { useMessage } from '../hooks/useMessage';
import copy from 'copy-to-clipboard';
import Button from '../components/Button';
import { CardContent, TextField, Typography } from '@mui/material';
import Card from '@mui/material/Card';
import LoginWrapper from '../modules/LoginWrapper/LoginWrapper';
import { defineMessages, useIntl } from 'react-intl';
import globalStore from '../lib/GlobalData';
import { TRANSLATE_ORIGIN_DATA } from '../constants/StoreKeys';
import TranslateHelp from '../modules/Help/translate';

const messages = defineMessages({
  copy: {
    id: 'btn.copy'
  },
  placeholder: {
    id: 'home.input-placeholder'
  },
  btn: {
    id: 'home.translate-btn'
  }
});
const Translate = () => {
  const [data, setData] = useState<string>(globalStore.get(TRANSLATE_ORIGIN_DATA));
  const { run, message, loading } = useMessage();
  const intl = useIntl();
  return (
    <>
      <Row
        gutter={12}
        style={{
          margin: 12
        }}
      >
        <Col
          span={12}
          style={{
            marginTop: 12
          }}
        >
          <div>
            <TextField
              label={intl.formatMessage(messages.placeholder)}
              onChange={e => {
                setData(e?.target?.value);
                globalStore.save(TRANSLATE_ORIGIN_DATA, e?.target?.value);
              }}
              rows={18}
              defaultValue={data}
              fullWidth
              variant="filled"
              multiline
            />
          </div>
        </Col>
        <Col span={12}>
          <div style={{ textAlign: 'center', marginBottom: 10 }}>
            <LoginWrapper>
              <Button
                variant="contained"
                loading={loading}
                type="primary"
                onClick={async () => {
                  data && (await run(data));
                }}
              >
                {intl.formatMessage(messages.btn)}
              </Button>
            </LoginWrapper>
          </div>
          <Card>
            <CardContent
              style={{
                minHeight: 340
              }}
            >
              <Typography sx={{ mb: 1.5 }} color="text.secondary">
                {message}
              </Typography>
            </CardContent>
          </Card>
          <div>
            <Button
              type="primary"
              onClick={() => {
                copy(message);
              }}
            >
              {intl.formatMessage(messages.copy)}
            </Button>
          </div>
        </Col>
      </Row>
    </>
  );
};

export default Translate;
