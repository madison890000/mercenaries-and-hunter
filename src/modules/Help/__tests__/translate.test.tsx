import React, {PropsWithChildren} from 'react';
import renderer from 'react-test-renderer';
import Translate from "../translate";
import {BrowserRouter as Router} from "react-router-dom";

const RouterWrapper: React.FC<PropsWithChildren> = ({children}) => {
    return <Router>{children}</Router>;
};

test('renders Translate', () => {
    const WithPaddingLayoutDom = renderer.create(
        <RouterWrapper>
            <Translate/>
        </RouterWrapper>
    ).toJSON();
    expect(WithPaddingLayoutDom).toMatchSnapshot();
});
