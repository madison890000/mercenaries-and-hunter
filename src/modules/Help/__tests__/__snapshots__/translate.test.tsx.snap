// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Translate 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  style={
    Object {
      "background": "var(--color-blue-9)",
      "margin": 30,
      "padding": 30,
      "textAlign": "left",
    }
  }
>
  <div
    className="ant-row ant-row-space-between css-dev-only-do-not-override-byeoj0"
    style={Object {}}
  >
    <div
      className="ant-col css-dev-only-do-not-override-byeoj0"
      style={Object {}}
    >
      <h2
        style={
          Object {
            "marginLeft": "20px",
          }
        }
      >
        AI translate
      </h2>
      <div
        style={
          Object {
            "fontStyle": "italic",
            "margin": 20,
            "textAlign": "left",
          }
        }
      >
        <div>
          不懂英语，不敢回复email？
        </div>
        <div>
          语法混乱，不敢回复email？
        </div>
        <div>
          语气不得体，不敢回复email？
        </div>
      </div>
      <div
        style={
          Object {
            "fontWeight": "bolder",
            "margin": 20,
            "textAlign": "left",
          }
        }
      >
        <div>
          一键翻译，告别错误英语！
        </div>
        <div
          style={
            Object {
              "marginTop": 20,
            }
          }
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            快去体验
          </button>
        </div>
      </div>
    </div>
    <div
      className="ant-col css-dev-only-do-not-override-byeoj0"
      style={Object {}}
    >
      <img
        height="320"
        src="/gif/email.gif"
      />
    </div>
  </div>
</div>
`;
